lib/sdam/server_selection.js

```diff
@@ -33,17 +33,19 @@ function maxStalenessReducer(readPreference, topologyDescription, servers) {
     return servers;
   }
 
-  const maxStaleness = readPreference.maxStalenessSeconds;
-  const maxStalenessVariance =
-    (topologyDescription.heartbeatFrequencyMS + IDLE_WRITE_PERIOD) / 1000;
-  if (maxStaleness < maxStalenessVariance) {
-    throw new MongoError(`maxStalenessSeconds must be at least ${maxStalenessVariance} seconds`);
-  }
+  if (topologyDescription.type !== TopologyType.Unknown) {
```

Comment:
had to do this because of: 
https://github.com/mongodb/specifications/blob/e56f5eceed7729f8b9b43a4a1f76c7e5840db49f/source/max-staleness/tests/Unknown/SmallMaxStaleness.yml#L1

Reply:
is this a change from another PR, or new code?

Reply:
this is new code to make that specific spec work in master

Reply:
looks like this requirement was added in [this spec commit](https://github.com/mongodb/specifications/commit/69e4225a20c3a53428b02afec8b9cbc9a5a3a2a0#diff-badcf416f8d59df7eeb0a22ba4ca4a9fR93) quite a while back

Reply:
I think my question is: why didn't we need this in 3.6? And I think the answer is because [this commit](https://github.com/mongodb/node-mongodb-native/commit/af57b578dd603faa7b66983232de2bc7e417dae1) was not ported to `master`. Can you please cherry-pick that one in and confirm it fixes the issue you were trying to solve here?